### PR TITLE
Feat video listing page

### PR DESCRIPTION
**Pull request by ianuragsharma**

### CODE REVIEW

File: src/components/navbar/Navbar.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+function Navbar() {
+  const [sidebar, setSidebar] = useState(true);
+  useLayoutEffect(() => {
+    function updateSize() {
+      window.innerWidth > 756 ? setSidebar(true) : setSidebar(false);
+    }
+    window.addEventListener("resize", updateSize);
+    updateSize();
+    return () => window.removeEventListener("resize", updateSize);
+  }, []);
```
Comment (rohanmathur91) [inline src/components/navbar/Navbar.jsx:17]: suggestion: put a comment explaining what is happening here

File: src/components/videoCard/VideoCard.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    tumbnailURL,
+    avatar,
+    category,
+  } = video;
+  const textOverflow = (title) => {
+    let newTitle = "";
+    if (title.length >= 75) {
+      newTitle = title.substring(0, 73) + "...";
+    }
+    return newTitle ? newTitle : title;
```
Comment (rohanmathur91) [inline src/components/videoCard/VideoCard.jsx:22]: you can handle this with CSS, any specific reason to do it this way?

File: src/context/video-context.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+              payload: [...videoResponse.data.videos],
+            });
+            videoDispatch({
+              type: "SET_CATEGORIES",
+              payload: [...categoriesResponse.data.categories],
+            });
+          }
+        } catch (e) {
+          console.error(" Error in fetching videos", e);
+        }
```
Comment (rohanmathur91) [inline src/context/video-context.js:33]: try to show error on UI using toast

**Review by rohanmathur91 — APPROVED** [review]
- Code is clean, added a few suggestions have a look.
- Overall looks good to merge.

File: src/components/videoCard/videoCard.css [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  background-color: var(--solid-black);
+  position: absolute;
+  right: 8px;
+  bottom: 1rem;
+}
+.video-uplodaded {
+  margin-left: auto;
+}
+.video-alt {
+  color: #7a7a7a;
```
Comment (ishitakeshawani) [inline src/components/videoCard/videoCard.css]: define variable for this color.

File: src/context/video-context.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import axios from "axios";
+const initialState = {
+  videos: [],
+  categories: [],
+  selectedCategory: "All",
+};
+
+const VideoContext = createContext();
+const VideoProvider = ({ children }) => {
+  const videoReducer = (videoState, { type, payload }) => {
```
Comment (ishitakeshawani) [inline src/context/video-context.js]: You can make folder reducer and put videoReducer in videoreducer file.

File: src/components/navbar/Navbar.jsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            </Link>
+            <Link to="/">
+              <h4 className="fw-500 text-xl ml-2">Underrated</h4>
+            </Link>
+          </li>
+          <div className="pt-1 sidebar-container">
+            {sidebarData.map((item, index) => {
+              return (
+                <li key={index} className="nav-text mt-1">
+                  <Link className="nav-text-link mx-2 text-xl" to={item.path}>
```
Comment (ishitakeshawani) [inline src/components/navbar/Navbar.jsx]: You can use NavLink here and can show active item.
https://v5.reactrouter.com/web/api/NavLink

**Review by ishitakeshawani — APPROVED** [review]
Added  few suggestions. Other things seems good.
Liked UI of home page.

File: src/components/navbar/Navbar.jsx (hunk 1 above)
Comment (ianuragsharma) [reply src/components/navbar/Navbar.jsx:17]: yeah sure.

File: src/components/videoCard/VideoCard.jsx (hunk 2 above)
Comment (ianuragsharma) [reply src/components/videoCard/VideoCard.jsx:22]: Yes text ellipsis works only for single line i wanted to do it for two lines actually.

File: src/context/video-context.js (hunk 3 above)
Comment (ianuragsharma) [reply src/context/video-context.js:33]: Will do that

File: src/components/videoCard/videoCard.css (hunk 4 above)
Comment (ianuragsharma) [reply src/components/videoCard/videoCard.css]: sure.

File: src/context/video-context.js (hunk 5 above)
Comment (ianuragsharma) [reply src/context/video-context.js]: okay

File: src/components/navbar/Navbar.jsx (hunk 6 above)
Comment (ianuragsharma) [reply src/components/navbar/Navbar.jsx]: will do that.

**Pull request merged**